lib/appcues_flutter.dart

```diff
@@ -314,11 +318,19 @@ class Appcues {
         .invokeMethod('didHandleURL', {'url': url.toString()});
   }
 
+  static void _listenToSemanticsUpdates(PipelineOwner pipelineOwner) {
+    var handle = pipelineOwner.ensureSemantics(
+        listener: () => _semanticsChanged(pipelineOwner));
+    _semanticsHandles.add(handle);
+    // recurse over the children of this PipelineOwner to ensure we
+    // set up listeners for each potential rendering tree
+    pipelineOwner.visitChildren((child) => _listenToSemanticsUpdates(child));
+  }
+
   // runs every time the SemanticsNode tree updates, capturing the known
   // layout information that can be used for Appcues element targeting
-  static void _semanticsChanged() {
-    var rootSemanticNode = RendererBinding
-        .instance.pipelineOwner.semanticsOwner?.rootSemanticsNode;
+  static void _semanticsChanged(PipelineOwner pipelineOwner) {
+    var rootSemanticNode = pipelineOwner.semanticsOwner?.rootSemanticsNode;
```

Comment:
here is a gotcha to consider: what if there are multiple PipelineOwner objects sending updates, in some multi view scenario - the last one wins. The last one will trigger the call to `setTargetElements` at the end of this function and dictate what target elements are visible for our appcues content.

In practice, it seems flutter mobile will only really have one RenderBinding/pipeline from what I've read, so this should not be an issue - I hope - and we'll only have one listener really sending updates here. Otherwise, maybe the current view that is taking the snapshot is the one sending the update and that will work as expected also? I'm not sure, but I hope. I don't see a way to handle multiple updates and know which ones are "active" and should be appended versus replaced, though, at this point.